<?xml version="1.0"?>
<robot  xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:property name="M_PI" value="3.1415926535897931"/>

  <xacro:macro name="cuboid_inertia" params="width height length mass">
    <inertia ixx="${mass * (height * height + length * length) / 12}"
             iyy="${mass * (width * width + length * length) / 12}"
             izz="${mass * (width * width + height * height) / 12}"
             ixy="0" iyz="0" ixz="0"/>
  </xacro:macro>

  <!-- length is along the y-axis! -->
  <xacro:macro name="cylinder_inertia" params="radius length mass">
    <inertia ixx="${mass * (3 * radius * radius + length * length) / 12}"
             iyy="${mass * radius* radius / 2}"
             izz="${mass * (3 * radius * radius + length * length) / 12}"
             ixy="0" iyz="0" ixz="0"/>
  </xacro:macro>

  <xacro:macro name="box_inertia" params="m x:=${x} y z">
    <inertia ixx="${m*(y*y+z*z)/12}" ixy="0" ixz="0" iyy="${m*(x*x+z*z)/12}" iyz="0" izz="${m*(x*x+z*z)/12}"/>
  </xacro:macro>


  <xacro:macro name="collision_sphere" params="radius xyz rpy">
    <collision>
      <origin rpy="${rpy}" xyz="${xyz}"/>
      <geometry>
        <sphere radius="${radius}"/>
      </geometry>
      <contact_coefficients mu="100000000.0" mu2="100000000.0" kp="100000.0" kd="100000.0" />
    </collision>
  </xacro:macro>


  <material name="pino">
     <color rgba="${158/255} ${22/255} ${22/255} 1.0"/>
  </material>

  <material name="gred">
     <color rgba="${0/255} ${175/255} ${175/255} 1.0"/>
  </material>
</robot>
